server/src/main/java/org/apache/druid/server/coordinator/duty/CompactSegments.java

```diff
@@ -234,12 +234,13 @@ private boolean cancelTaskIfGranularityChanged(
     Granularity configuredSegmentGranularity = dataSourceCompactionConfig.getGranularitySpec()
                                                                          .getSegmentGranularity();
     Granularity taskSegmentGranularity = compactionTaskQuery.getGranularitySpec().getSegmentGranularity();
```

Comment:
The fix makes sense. It seems that the members and getters in `UserCompactionTaskGranularityConfig` are missing `@Nullable` annotations. Could we also update those?

Also, for this bug fix, it'll be good to update an existing test for in `CompactSegmentsTest` or another relevant test to pass null values to the compaction task query.

Reply:
Some of the `@Nullable` annotations are being added in another PR #16291 , so I skipped them here.
Okay if we add the tests separately? I want to merge this PR and not wait for another CI build.

Reply:
Works for me 👍